lib/LANraragi/Model/Tankoubon.pm

```diff
@@ -118,8 +181,29 @@ sub get_tankoubon ( $tank_id, $fulldata = 0, $page = 0 ) {
     my @limit = split( ' ', "LIMIT " . ( $keysperpage * $page ) . " $keysperpage" );
 
     # Get name
-    my @name = $redis->zrangebyscore( $tank_id, 0, 0, qw{LIMIT 0 1} );
-    $tank{name} = redis_decode( $name[0] );
+    my @name = $redis->zrangebyscore( $tank_id, $TANK_METADATA{"name"}, $TANK_METADATA{"name"}, qw{LIMIT 0 1} );
```

Comment:
In a similar way, this can probably be refactored into an individual sub.  

(And now that I'm thinking of it, couldn't you do a single zrangebyscore from 0 to -5 to get all the metadata items at once?)

Reply:
Supposing we do not increase the amount of metadata in the future, yes, we could get all in one harcoded call. Or maybe even better I can use the dict of keys to get the lower score... Will try a few things.

Reply:
Tried to do this, but the fact that a sorted set can have multiple values in the same score is opening the door to many bugs if someone managed to mess with the update_tankoubon EP. Basically it grabs all the elements within those scores, and then matches the names using the dict of keys, but if one scores has multiple elements, it causes a mismatch. It could be added that the zrangebyscore returns the elements with their scores to try to manage that, but sounds like a lot of things when the number of lines we are replacing will not decrease.

For the moment I added the method that resolves this inside tankoubon_tools method, but I think is a good idea for you to give it a look before implementing it.

Reply:
So, I asked our friend ChatGPT for a way to fix the problem with this new method, the solution doubles the complexity, but at least ensure we get the pairings correctly(There might still be bugs, but that will get there little by little). Also added a new method to clean the result of get_tankoubon, since the result was generating some thrash fields.